### PR TITLE
Add test to check reporting not enough replicas

### PR DESCRIPTION
**Pull request by stlaz**
Currently fails because it can't find `authentications.operator.openshift.io "cluster" not found` even though it exists?

### CODE REVIEW

File: test/e2e/conditions_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		return compareConditions(expectedCondition, actualCondition), nil
+	})
+}
+
+func compareConditions(c1, c2 *operatorv1.OperatorCondition) bool {
+	if c1 == c2 { // both are nil or the same pointer
+		return true
+	}
+
+	return c1.Type == c2.Type && c1.Status == c2.Status &&
```
Comment (vareti) [inline test/e2e/conditions_test.go]: do we also need to consider the case where only one of `c1` or `c2` is nil?

Comment (stlaz) [reply test/e2e/conditions_test.go]: yes, should be fixed now

**Comment by marun** [comment]
@stlaz Is an e2e necessary, or would a unit test of `checkDeploymentReady` suffice?

**Comment by stlaz** [comment]
I think testing conditions makes more sense in what we could call the real world

**Comment by marun** [comment]
> I think testing conditions makes more sense in what we could call the real world

The only thing this e2e is validating that a unit test wouldn't is retrieval of a deployment. I'm not exactly sure why that is something valuable to test given that failure to retrieve the deployment will result in a degraded condition that would be caught without this e2e.

**Comment by stlaz** [comment]
/close
this got stale, if we need it, we could perhaps do it better as Maru suggested

**Pull request closed without merging**